src/allotropy/allotrope/schema_parser/model_class_editor.py

```diff
@@ -54,25 +77,333 @@ def get_manifest_from_schema_path(schema_path: str) -> str:
+        self.parent_class_names.remove(parent_class.class_name)
+
+        self.fields |= parent_class.fields
+        for field_name in parent_class.field_name_order:
+            if field_name not in self.field_name_order:
+                self.field_name_order.append(field_name)
+
+        return DataClassLines.create(
+            self.class_name,
+            self.parent_class_names,
+            self.fields,
+            self.field_name_order,
+            self.is_frozen,
+        )
+
+    def should_merge(self, other: DataClassLines) -> bool:
+        # parent classes must match
+        if not set(self.parent_class_names) == set(other.parent_class_names):
+            return False
+        # There must be some overlapping fields with the same values
+        if not any(
+            self.fields[name].contents == other.fields[name].contents
+            for name in self.fields.keys() & other.fields.keys()
+        ):
+            return False
+        # Fields unique to one class must be optional.
+        if any(
+            self.fields[name].is_required
+            for name in self.fields.keys() - other.fields.keys()
+        ):
+            return False
+        if any(
+            other.fields[name].is_required
+            for name in other.fields.keys() - self.fields.keys()
+        ):
+            return False
+        # Shared fields must agree on whether they are required
+        if not all(
+            self.fields[name].can_merge(other.fields[name])
+            for name in self.fields.keys() & other.fields.keys()
+        ):
+            return False
+
+        return True
+
+    def merge_similar(self, other: DataClassLines) -> DataClassLines:
+        # Merge another class that has similar fields with this one.
+
+        # Add fields from the other class to this one.
+        for field_name in other.fields:
+            if field_name not in self.fields:
+                self.fields[field_name] = other.fields[field_name]
+                self.field_name_order.append(field_name)
+
+        # Merge fields by combining types into a single union.
+        for field_name in self.fields.keys() & other.fields.keys():
+            if self.fields[field_name].contents == other.fields[field_name].contents:
+                continue
+            self.fields[field_name] = self.fields[field_name].merge(
+                other.fields[field_name]
+            )
+
+        return DataClassLines.create(
+            self.class_name,
+            self.parent_class_names,
+            self.fields,
+            self.field_name_order,
+            self.is_frozen,
+        )
+
+
+def create_class_lines(lines: list[str]) -> ClassLines:
+    is_dataclass = lines[0].startswith("@dataclass")
+
+    # Get the lines that are the class description, including the name and parent classes.
+    desc_start = 1 if is_dataclass else 0
+    desc_end = desc_start
+    while desc_end < len(lines) and ":" not in lines[desc_end]:
```

Comment:
Added a comment about multiline to explain.

For comment/decorator, the answer is a hopeful no, since we are only dealing with auto-generated code here, and I haven't seen anything like that. If we do later, it should fail pretty obviously :D